src/documents/firstDiscoveryServer.md

```diff
@@ -0,0 +1,365 @@
+        </tr>
+        <tr>
+            <td>
+                <code>"preferencesConfig.authentication.client_secret"</code>
+            </td>
+            <td>
+                <code>GPII_OAUTH2_AUTH_CLIENT_SECRET</code>
+            </td>
+            <td>
+                The client secret registered with the OAuth2 server.
+                Used to securely identify the client with the OAuth2.
+                Should be kept confidentially.
+            </td>
+            <td>
+                "client_secret_first_discovery"
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+### oauth2.json ###
+
+Builds on [environment.json](https://github.com/GPII/first-discovery-server/tree/master/src/config/environment.json), setting concrete values to configuration of the Oauth2 server that are known to exists for the GPII security layer.
+
+(see: [oauth2.json](https://github.com/GPII/first-discovery-server/tree/master/src/config/oauth2.json))
+
+<table>
+    <thead>
+        <tr>
+            <th>Option</th>
+            <th>Description</th>
+            <th>Value</th>
+        <tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>
+                <code>"preferencesConfig.securityServer.paths.token"</code>
+            </td>
+            <td>
+                The path to the resource for requesting an access token
+            </td>
+            <td>
+                "/access_token"
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <code>"preferencesConfig.securityServer.paths.preferences"</code>
+            </td>
+            <td>
+                The path to the resource for creating a preference set.
+                A query parameter can be added to provide the view/ontology that the
+                preferences are to be stored in.
+            </td>
+            <td>
+                "/add-preferences?view=%view"
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <code>"preferencesConfig.authentication.grant_type"</code>
+            </td>
+            <td>
+                The grant type supported by the OAuth2 server.
+            </td>
+            <td>
+                "client_credentials"
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <code>"preferencesConfig.authentication.scope"</code>
+            </td>
+            <td>
+                The level of permissions that are being requested.
+            </td>
+            <td>
+                "add_preferences"
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+### vagrant.json ###
+
+Builds on [environment.json](https://github.com/GPII/first-discovery-server/tree/master/src/config/environment.json) and [oauth2.json](https://github.com/GPII/first-discovery-server/tree/master/src/config/oauth2.json), setting concrete values to configuration of the Oauth2 server that are expected when being run via the Vagrant VM. This is typically only used for development.
+
```

Comment:
"... when being run by the Vagrant VM." What vagrant vm? Is this something the reader is expected to know about? I suspect this should either say something more like "when being run in a Vagrant VM" or if this is referencing something specific, give some information about it.